superset-frontend/src/explore/components/DatasourcePanel/index.tsx

```diff
@@ -289,6 +301,22 @@ export default function DataSourcePanel({
     [lists.columns, showAllColumns],
   );
 
+  const showInfoboxCheck = () => {
+    if (sessionStorage.getItem('showInfobox') === 'false') return false;
+    return true;
+  };
+
+  const datasourceTypeCheck = () => {
```

Comment:
We could rewrite as: 
`const datasourceTypeCheck = datasource.type == DatasourceType.Dataset. . . .,etc` 

and on checking it would return true or false. That way you don't have to write the if statement at all.

Reply:
That's even better! Changed in [`this commit`](https://github.com/apache/superset/pull/19855/commits/9cc44abcf0e89874d2094ac1fe7b39a912b4ae2c).